src/libraries/System.Net.NameResolution/tests/PalTests/NameResolutionPalTests.cs

```diff
@@ -204,11 +213,145 @@ public void TryGetNameInfo_LocalHost_IPv6_TryGetAddrInfo(bool justAddresses)
             Assert.NotNull(addresses);
         }
 
+        [Fact]
+        public void HostName_NotNull()
+        {
+            Assert.NotNull(NameResolutionPal.GetHostName());
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task GetAddrInfoAsync_LocalHost(bool justAddresses)
+        {
+            if (!NameResolutionPal.SupportsGetAddrInfoAsync)
+            {
+                return;
+            }
+
+            if (justAddresses)
+            {
+                IPAddress[] addresses = await ((Task<IPAddress[]>)NameResolutionPal.GetAddrInfoAsync("localhost", justAddresses)).ConfigureAwait(false);
+
+                Assert.NotNull(addresses);
+                Assert.True(addresses.Length > 0);
+            }
+            else
+            {
+                IPHostEntry hostEntry = await ((Task<IPHostEntry>)NameResolutionPal.GetAddrInfoAsync("localhost", justAddresses)).ConfigureAwait(false);
+
+                Assert.NotNull(hostEntry);
+                Assert.True(hostEntry.AddressList.Length > 0);
+            }
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        [OuterLoop("Uses external server")]
+        public async Task GetAddrInfoAsync_HostName(bool justAddresses)
+        {
+            if (!NameResolutionPal.SupportsGetAddrInfoAsync)
+            {
+                return;
+            }
+
+            string hostName = NameResolutionPal.GetHostName();
+            Assert.NotNull(hostName);
+
+            Task task = NameResolutionPal.GetAddrInfoAsync(hostName, justAddresses);
+
+            try
+            {
+                await task.ConfigureAwait(false);
+            }
+            catch (SocketException ex)
+            {
+                SocketError error = ex.SocketErrorCode;
+
+                if (error == SocketError.HostNotFound && (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.OSX)))
+                {
+                    // On Unix, we are not guaranteed to be able to resove the local host. The ability to do so depends on the
+                    // machine configurations, which varies by distro and is often inconsistent.
+                    return;
+                }
+
+                throw;
+            }
+
+            if (justAddresses)
+            {
+                IPAddress[] addresses = ((Task<IPAddress[]>)task).Result;
+
+                Assert.NotNull(addresses);
+                Assert.True(addresses.Length > 0);
+            }
+            else
+            {
+                IPHostEntry hostEntry = ((Task<IPHostEntry>)task).Result;
+
+                Assert.NotNull(hostEntry);
+                Assert.True(hostEntry.AddressList.Length > 0);
+            }
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task GetAddrInfoAsync_ExternalHost(bool justAddresses)
```

Comment:
Should these tests be marked with `Outerloop`?

Reply:
I probably would not at this point. Since all DNS tests depend on external server, we would need to make all of them outer loop IMHO. For now, I would assume that base resolution work and I would leave it as inner loop. was thinking while back about running batch of resolutions to make sure DNS work and perhaps prime caches so test runs are more predictable. I think we can leave is is for now and react if we see CI failures.